docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,667 @@
+      "value": 200,
+      "mask": 200
+    },
+    "apply_actions": [
+      <will be described later>
+    ],
+    "write_actions": [
+      <will be described later>
+    ]
+  }
+}
+```
+
+</details>
+
+### Validate rules, meters, lags and groups installed on the switch v2
+
+`GET https://{host}/v2/switches/{switch_id}/validate`
+
+**By Default:**
+
+* `flow info` is set of values: for groups - `flow_id, flow_path`, for meters and rules - `flow_id, flow_path, y_flow_id`.
+* without any query params response contains v2 info about groups, lags, meters and rules (with flow_info).
+* groups[excess], meters[excess] and rule[excess] do not contain flow info.
+* for groups, lags, meters and rules flow info will be presented in misconfigured[expected] field.
+* misconfigured[id] is 
+  * meter_id for meters
+  * logical_port_id for logical_ports
+  * group_id for groups
+  * (priority + match).toString() for rules // may change during api implementation
+
+Response payload
+
+```json
+{
+  "as_expected": true,
+  "groups": {
+    "as_expected": true,
+    "excess": [
+      <v2GroupInfo> // without flow_id and flow_path
+    ],
+    "missing": [
+      <v2GroupInfo>
+    ],
+    "proper": [
+      <v2GroupInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "12123213",
+        "expected": {
+          <v2GroupInfo>
+        },
+        "discrepancies": {
+          <v2GroupInfo>
+        }
+      }
+    ]
+  },
+  "logical_ports": {
+    "as_expected": true,
+    "excess": [
+      <v2LogicalPortInfo>
+    ],
+    "missing": [
+      <v2LogicalPortInfo>
+    ],
+    "proper": [
+      <v2LogicalPortInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "123123",
+        "expected": {
+          <v2LogicalPortInfo>
+        },
+        "discrepancies": {
+          <v2LogicalPortInfo>
+        }
+      }
+    ]
+  },
+  "meters": {
+    "as_expected": true,
+    "excess": [
+      <v2MeterInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2MeterInfo>
+    ],
+    "proper": [
+      <v2MeterInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "1232131",
+        "expected": {
+          <v2MeterInfo>
+        },
+        "discrepancies": {
+          <v2MeterInfo>
+        }
+      }
+    ]
+  },
+  "rules": {
+    "as_expected": true,
+    "excess": [
+      <v2RuleInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2RuleInfo>
+    ],
+    "proper": [
+      <v2RuleInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "123123123",
+        "expected": {
+          <v2RuleInfo>
+        },
+        "discrepancies": {
+          <v2RuleInfo>
+        }
+      }
+    ]
+  }
+}
+```
+
+### Validate rules, meters, lags and groups installed on the switch with filters
+
+### Requesting validation of specific switch objects/subjects.
+
+Request params
+
+To get only group info in response use query param `groups`, other fields will be omitted.
```

Comment:
It will validate all fields/targets but filter out result to only requested set of field, or it will validate only requested fields/tartget? We should mention this behaviour here.

Reply:
It will validate all fields/targets but filter out result to only requested set of field - I'll add it in next PR